File: backend/Backend/Models/Game/GameState.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

public class GameState {
  [JsonPropertyName("type")]
  public required GameStateType StateType { get; set; }

  [JsonPropertyName("question")]
  public GameQuestion? Question { get; set; }

  [JsonPropertyName("answer")]
  public GameAnswer? Answer { get; set; }

  [JsonPropertyName("results")]
  public GameResults? Results { get; set; }

  [JsonPropertyName("nextQuestionInd")]
  public int NextQuestionInd { get; set; } = 0;

  [JsonPropertyName("participantsAnswers")]
  public Dictionary<string, GameAnswer>? ParticipantsAnswers { get; set; }
}

File: frontend/src/models/game/GameState.ts
import GameStateType from "./GameStateType";
import GameQuestion from "./GameQuestion";
import GameAnswer from "./GameAnswer";
import GameResults from "./GameResults";

export default interface GameState {
  type: GameStateType;
  question?: GameQuestion;
  answer?: GameAnswer;
  results?: GameResults;
  nextQuestionInd?: number;
  participantsAnswers?: Record<string, GameAnswer>;
}